torchgeo/datasets/geo.py

```diff
@@ -675,21 +675,6 @@ def __getitem__(self, query: BoundingBox) -> Dict[str, Any]:
 
         return sample
 
-    def plot(self, data: Tensor) -> None:
```

Comment:
Since CBF was the only `VectorDataset`, and since it overrides the `plot` method, this code was no longer being tested. We have a few options:

1. Remove `VectorDataset.plot`
2. Keep it, update the signature to take in a sample, and use a useful default
3. Keep it, update the signature to take in a sample, and raise NotImplementedError
4. Turn it into an abstract method and require all subclasses to implement it

The benefit of 1 is that it is the least code. The benefit of 2-4 is that they add mypy checks to ensure that all subclasses use the same signature. The benefit of 4 is that it prevents someone from adding a new subclass that doesn't implement plot. The downside of 4 is that it means you can't instantiate a `VectorDataset` directly.

I forget where we discussed this already, but I have a slight preference for 4 because it seems like "the right thing to do", and @calebrob6 has a slight preference for 3 because it allows you to use `VectorDataset` directly without creating a subclass. I think 1 and 2 are valid options too. Interested in hearing other opinions on this one.

Reply:
I wonder if our plotting code has enough in common that we could define one or two `plot` methods in `VisionDataset` and `GeoDataset` and then have each subclass call super with the appropriate arguments.

Reply:
Does anyone have any opinions on this? We should make this decision so we can finish adding plot methods to all GeoDatasets.

Reply:
> I wonder if our plotting code has enough in common that we could define one or two plot methods in VisionDataset and GeoDataset and then have each subclass call super with the appropriate arguments.

We do have a good bit of duplication, but I think a utility function would be better for the average case as there are some really whacky plots.